### DISCUSSION TITLE
Using dynamic props with ui.editor

### DISCUSSION THREAD

**Discussion by thetableman** [zauberzeug/nicegui#2029] [Q&A]
### Question

I'm planning to use the `ui.editor` element with additional toolbar settings as demonstrated in the Quasar 'Kitchen Sink' example here https://quasar.dev/vue-components/editor/#example--kitchen-sink

Consider the below based on the example from the NiceGUI documentation and the above Quasar link. When passing these props the `ui.editor` seems to ignore them and just displays the default options.

```py
from nicegui import ui

_definitions = """:definitions="{bold: {label: 'Bold', icon: null, tip: 'My bold tooltip'} }" """
_toolbar = """:toolbar="[[{label: $q.lang.editor.align,icon: $q.iconSet.editor.align,fixedLabel: true,options: ['left', 'center', 'right', 'justify']}],['quote', 'unordered', 'ordered', 'outdent', 'indent'], ['bold', 'italic', 'strike', 'underline', 'subscript', 'superscript']]" """

editor = ui.editor(placeholder='Type something here').props(_toolbar + _definitions)
ui.markdown().bind_content_from(editor, 'value', backward=lambda v: f'HTML code:\n```\n{v}\n```')

ui.run()
```

However, in this more simple example with fewer toolbar settings changed, the toolbar options display as expected.

```py
from nicegui import ui

_toolbar = """:toolbar="[['quote', 'unordered', 'ordered', 'outdent', 'indent'], ['bold', 'italic', 'strike', 'underline', 'subscript', 'superscript']]" """

editor = ui.editor(placeholder='Type something here').props(_toolbar)
ui.markdown().bind_content_from(editor, 'value', backward=lambda v: f'HTML code:\n```\n{v}\n```')

ui.run()
```

Additionally, toolbar options like `Unordered List`, `Ordered List` and `Blockquote` all create the relevant HTML code but fail to display correctly in the `ui.editor` element.

![image](https://github.com/zauberzeug/nicegui/assets/78121198/7465d076-7fc5-4849-b79b-ddf08f9821b4)

Has anyone had any success implementing more complex toolbar options?

**Comment by falkoschindler** [comment]
Interesting! The JavaScript console says:

```
Error while converting :toolbar attribute to function: ReferenceError: $q is not defined
Error while converting :definitions attribute to function: SyntaxError: Unexpected token ':'
```

Apparently there is no `$q`, but you can use `Quasar` instead. However, there is no `Quasar.iconSet.editor`...
And I have no clue why it's complaining about the colon ":".

Anyway, maybe this is a starting point for fixing the props. The JavaScript console should help with debugging quite a bit.

**Reply by thetableman** [reply]
I managed to get a working version by excluding the `$q.iconSet.editor`... and replacing with static labels and icons, this works fine for my purpose. 

```py
from nicegui import ui

_toolbar = """:toolbar="[
    [{label: 'Align', icon: 'subject', fixedLabel: true, list: 'only-icons', options: ['left', 'center', 'right', 'justify']}],
    [{label: 'Size', icon: 'format_size', fixedLabel: true, fixedIcon: true, list: 'no-icons', options: ['size-1','size-2','size-3','size-4','size-5','size-6','size-7']}],
    ['bold', 'italic', 'strike', 'underline', 'subscript', 'superscript'],
    ] " """.replace('\n','')

editor = ui.editor(placeholder='Type something here').props(_toolbar)
ui.markdown().bind_content_from(editor, 'value', backward=lambda v: f'HTML code:\n```\n{v}\n```')

ui.run()
```

However, despite success with creating `Align` and `Size` menus this way it doesn't work with `Font` which returns `TypeError: o is undefined`. This might be because the `:fonts=`... prop is missing but when I do pass this I get the same issue with the ":" as mentioned above.

@falkoschindler Any thoughts on why `Unordered List`, `Ordered List` and `Blockquote` don't display in the editor?

**Reply by falkoschindler** [reply]
Oh, Tailwind seems to interfere with the formatting. If you run with `ui.run(tailwind=False)`, blockquotes and lists get formatted correctly. I'll need to think about how we can fix that...

**Reply by thetableman** [reply]
I noticed a similar issue when using `ui.markdown`, an `Unordered List` displays but an `Ordered List` doesn't:
```py
from nicegui import ui

textarea = ui.textarea('')
ui.markdown().bind_content_from(textarea, 'value')

ui.run()
```
![image](https://github.com/zauberzeug/nicegui/assets/78121198/89cebdc0-825e-4af8-8e38-0bdf517b7373)

**Reply by falkoschindler** [reply]
Thanks, @thetableman! I created a new issue to tackle this problem: #2045.
